### PR TITLE
Drop explicit support for python2

### PR DESCRIPTION
**Pull request by alurban**
This PR drops all support for python-2.7, now that we've been running stably in python-3.7:

* Remove imports from `__future__`
* Replace imports from `six` with their python3 equivalents
* Enforce system python versions >= 3.5

This is part of an extensive purge of python2 across our detchar software stack, see gwdetchar/hveto#126 and gwpy/gwsumm#267.

cc @duncanmmacleod

### CODE REVIEW

File: bin/gwdetchar-scattering [hunk 1]
```diff
@@ -25,14 +25,11 @@ time please use the command-line module:
 `python -m gwdetchar.scattering --help`
 """
 
-from __future__ import division
-
-import os.path
 import re
 import random
+import os.path
```
Comment (duncanmmacleod) [inline bin/gwdetchar-scattering]: Why was this moved?

File: gwdetchar/utils.py [hunk 2]
```diff
@@ -21,15 +21,10 @@
...
-    from io import StringIO
-    from html.parser import HTMLParser
-except:  # python 2.7
-    from cStringIO import StringIO
-    from HTMLParser import HTMLParser
-
 import numpy
+from io import StringIO
+from functools import partial
+from html.parser import HTMLParser
```
Comment (duncanmmacleod) [inline gwdetchar/utils.py]: Can you move these about the numpy import, since they are standard library modules?

Comment (duncanmmacleod) [inline gwdetchar/utils.py]: Can you move these above the `numpy` import, since they are standard library modules?

File: setup.py [hunk 3]
```diff
@@ -31,6 +29,10 @@
 # local setup utilities
 from _setup_utils import CMDCLASS as cmdclass
 
+# enforce python version
+if sys.version < '3.5':
+    raise ImportError("Python versions older than 3.5 are not supported.")
+
```
Comment (duncanmmacleod) [inline setup.py]: I don't think we need to include this.

File: versioneer.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 * https://github.com/warner/python-versioneer
 * Brian Warner
 * License: Public Domain
-* Compatible With: python2.6, 2.7, 3.3, 3.4, 3.5, and pypy
+* Compatible With: python3.5+, conda, and pypy
```
Comment (duncanmmacleod) [inline versioneer.py]: Please revert this change, since this is just a vendored copy of `versioneer`.

```suggestion
* Compatible With: python2.6, 2.7, 3.3, 3.4, 3.5, and pypy
```

File: bin/gwdetchar-scattering (hunk 1 above)
Comment (alurban) [reply bin/gwdetchar-scattering]: @duncanmmacleod, mostly it's because I'm obsessive about the flow of text width in import statements -- I like to go from narrow to wide, for cleanliness of presentation

File: gwdetchar/utils.py (hunk 2 above)
Comment (alurban) [reply gwdetchar/utils.py]: @duncanmmacleod, this is done in the last commit.

**Comment by alurban** [comment]
@duncanmmacleod, thanks for your review, I've addressed your comments above
